framework/src/outputs/CSV.C

```diff
@@ -173,6 +174,17 @@ CSV::getVectorPostprocessorFileName(const std::string & vpp_name,
       file_name << '_' << std::setw(_padding) << std::setprecision(0) << std::setfill('0')
                 << std::right << _linear_iter;
     }
+
+    // These lines enable CSV per-optimization iteration output (Optimization module)
+
+    if (_app.getExecutioner()->getIterationOutputFlag() &&
+        _current_execute_flag == EXEC_TIMESTEP_END)
```

Comment:
Does the optimization module have its own execute on flags?

Reply:
Yes, it has its own execute on flags: `EXEC_FORWARD`, `EXEC_ADJOINT`, `EXEC_ADJOINT_TIMESTEP_BEGIN`,...